src/components/table/helpers/mixin-filtering.js

```diff
@@ -20,77 +20,96 @@ export default {
     filterFunction: {
       type: Function,
       default: null
+    },
+    noLocalFiltering: {
+      type: Boolean,
+      default: false
     }
   },
   data() {
     return {
-      // Flag for displaying which empty slot to show, and for some event triggering.
+      // Flag for displaying which empty slot to show and some event triggering
       isFiltered: false
     }
   },
   computed: {
     localFiltering() {
-      return this.hasProvider ? !!this.noProviderFiltering : true
+      return this.hasProvider ? !!this.noProviderFiltering : !this.noLocalFiltering
     },
+    // For watching changes to `filteredItems` vs `localItems`
     filteredCheck() {
-      // For watching changes to filteredItems vs localItems
       return {
         filteredItems: this.filteredItems,
         localItems: this.localItems,
         localFilter: this.localFilter
       }
     },
+    // Sanitized/normalized version of filter prop
     localFilter() {
-      // Returns a sanitized/normalized version of filter prop
+      // Early exit when filtering is disabled
+      if (!this.localFiltering) {
+        return ''
+      }
+
+      // Deprecate setting prop filter to a function
+      // `localFilterFn` will contain the correct function ref
       if (isFunction(this.filter)) {
-        // this.localFilterFn will contain the correct function ref.
-        // Deprecate setting prop filter to a function
         /* istanbul ignore next */
         return ''
-      } else if (
-        this.localFiltering &&
-        !isFunction(this.filterFunction) &&
-        !(isString(this.filter) || isRegExp(this.filter))
-      ) {
-        // Using internal filter function, which only accepts string or regexp at the moment
+      }
+
+      // Using internal filter function, which only accepts string or RegExp
+      if (!isFunction(this.filterFunction) && !(isString(this.filter) || isRegExp(this.filter))) {
         return ''
-      } else {
-        // Could be a string, object or array, as needed by external filter function
-        // We use `cloneDeep` to ensure we have a new copy of an object or array
-        // without Vue reactive observers.
-        return cloneDeep(this.filter)
       }
+
+      // Could be a string, object or array, as needed by external filter function
+      // We use `cloneDeep` to ensure we have a new copy of an object or array
+      // without Vue reactive observers
+      return cloneDeep(this.filter)
     },
+    // Sanitized/normalize filter-function prop
     localFilterFn() {
-      let filter = this.filter
-      let filterFn = this.filterFunction
-      // Sanitized/normalize filter-function prop
```

Comment:
The two lets were here to ensure both props were reactively read before testing any conditions.